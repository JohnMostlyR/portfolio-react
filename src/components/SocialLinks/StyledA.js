import styled from 'styled-components';

const StyledA = styled.a`
  color: #575756;
  text-decoration: none;
`;

export default StyledA;