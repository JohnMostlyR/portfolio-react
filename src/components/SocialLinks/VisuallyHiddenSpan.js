import styled from 'styled-components';

import {visuallyHidden} from '../../styles/mixins';

const VisuallyHiddenSpan = styled.span`
  ${visuallyHidden}
`;

export default VisuallyHiddenSpan;