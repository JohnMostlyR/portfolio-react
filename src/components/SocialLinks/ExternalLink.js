import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

import StyledA from './StyledA';
import VisuallyHiddenSpan from './VisuallyHiddenSpan'

const ExternalLink = (props) => (
    <StyledA href={props.href} target="_blank" rel="noopener noreferrer">
      <FontAwesome name={props.faIcon}/><VisuallyHiddenSpan>{props.description}</VisuallyHiddenSpan>
    </StyledA>
);

ExternalLink.propTypes = {
  description: PropTypes.string,
  href: PropTypes.string,
  faIcon: PropTypes.string,
};

ExternalLink.defaultProps = {
  faIcon: 'meh-o',
  description: 'ExternalLink to',
  href: '#',
};

export default ExternalLink;