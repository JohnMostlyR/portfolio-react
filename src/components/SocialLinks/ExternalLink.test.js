import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { mount } from 'enzyme';
import FontAwesome from 'react-fontawesome';

import ExternalLink from './ExternalLink';
import StyledA from './StyledA';
import VisuallyHiddenSpan from './VisuallyHiddenSpan'

describe('ExternalLink', () => {
  const DESCRIPTION = 'Follow me';
  const EXTERNAL_LINK = 'https://some-site.com';
  const ICON = 'check-square-o';

  let wrapper;

  beforeEach(() => {
    wrapper = mount(
        <ExternalLink faIcon={ICON}
                      description={DESCRIPTION}
                      href={EXTERNAL_LINK}
        />,
    );
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(<ExternalLink/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have a `styled.a` component', () => {
    expect(
        wrapper.find(StyledA)
    ).toHaveLength(1);
  });

  it('Should have a `FontAwesome` component', () => {
    expect(
        wrapper.find(FontAwesome)
    ).toHaveLength(1);
  });

  it('Should have a `VisuallyHiddenSpan` component', () => {
    expect(
        wrapper.find(VisuallyHiddenSpan)
    ).toHaveLength(1);
  });

  describe('Anker element', () => {
    beforeEach(() => {
      wrapper = mount(
          <ExternalLink faIcon={ICON}
                        description={DESCRIPTION}
                        href={EXTERNAL_LINK}
          />,
      );
    });

    it('Should have an `href` property', () => {
      const A = wrapper.find(`a[href="${EXTERNAL_LINK}"]`);
      expect(A).toHaveLength(1);
    });

    it('Should have a `target` property set to `_blank`', () => {
      const A = wrapper.find(`a[href="${EXTERNAL_LINK}"]`);
      expect(A.props().target).toBe('_blank');
    });

    it('Should have a `rel` property set to `noopener noreferrer`', () => {
      const A = wrapper.find(`a[href="${EXTERNAL_LINK}"]`);
      expect(A.props().rel).toBe('noopener noreferrer');
    });
  });

  describe('FontAwesome component', () => {
    it('Should get the `name` property', () => {
      const FONT_AWESOME = wrapper.find(FontAwesome);
      expect(FONT_AWESOME.props().name).toBe(ICON);
    });
  });

  describe('VisuallyHiddenSpan component', () => {
    it('Should get the correct description', () => {
      const SPAN = wrapper.find(VisuallyHiddenSpan);
      expect(SPAN.text()).toBe(DESCRIPTION);
    });
  });
});