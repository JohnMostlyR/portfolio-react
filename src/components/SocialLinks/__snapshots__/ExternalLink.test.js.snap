// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink Should render and match the snapshot 1`] = `
.c0 {
  color: #575756;
  text-decoration: none;
}

.c1 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  padding: 0 !important;
  border: 0 !important;
  overflow: hidden !important;
  -webkit-clip: rect(0 0 0 0) !important;
  clip: rect(0 0 0 0) !important;
}

<a
  className="c0"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    aria-hidden={true}
    className="fa fa-meh-o"
  />
  <span
    className="c1"
  >
    ExternalLink to
  </span>
</a>
`;
